GenderPayGap.WebUI/Views/Viewing/EmployerDetails/Employer.cshtml

```diff
@@ -307,6 +334,55 @@
                 }
             });
         }());
+        
+        var parentForSelect = document.getElementById("filteringArea");
+        
+        var selectLabel = document.createElement("label");
+        selectLabel.className = "govuk-label";
+        selectLabel.for = "yearFilter";
+        selectLabel.textContent = "Filter for a reporting period";
+        
+        var elements = document.getElementsByClassName("yearTextForFiltering");
+        
+        var selectList = document.createElement("select");
+        selectList.className = "govuk-select govuk-!-margin-bottom-3";
+        selectList.id = "filterYearSelect";
+        selectList.addEventListener("change",function (e){
+            for (var i = 0; i < elements.length; i++){
+                var tableRow = elements[i];
+                while (tableRow.tagName.toLowerCase() !== "tr"){
+                    tableRow = tableRow.parentElement;    
+                }
+                tableRow.hidden = true;
+                
+                if (elements[i].innerHTML.includes(e.target.value) || e.target.value === "Show All"){
+                    tableRow.hidden = false;
+                }
+            }
+        }) 
+        
+        var selectHint = document.createElement("div");
+        selectHint.className = "govuk-hint";
+        selectHint.textContent = "For example, if the snapshot date is 5 April 2017 but the deadline is 4 April 2018 then the reporting period is 2017 - 2018.";
+       
+        parentForSelect.appendChild(selectLabel);
+        parentForSelect.appendChild(selectList);
+        parentForSelect.appendChild(selectHint);
 
+        var showAll = document.createElement("option");
+        showAll.value = "Show All";
+        showAll.text = "Show All";
+        showAll.selected = true;
+        selectList.appendChild(showAll);
+        
+        var arrayOfYears = JSON.parse(document.getElementById("selectOptionsFromReports").value);
+        for (var i = 1; i < arrayOfYears.length; i++){

```

Comment:
Is `i = 1` deliberate?

Reply:
It was intentional but incorrect, I was offsetting for the "show all" but that's not required.